src/server/pachyderm_test.go

```diff
@@ -10534,3 +10537,86 @@ func TestMissingSecretFailure(t *testing.T) {
 		return nil
 	})
 }
+
+type rowCountResults struct {
+	IdxScan    int    `json:"idx_scan"`
+	NLiveTup   int    `json:"n_live_tup"`
+	RelName    string `json:"relname"`
+	SchemaName string `json:"schemaname"`
+	SeqScan    int    `json:"seq_scan"`
+}
+
+func TestDatabaseStats(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	repoName := tu.UniqueString("TestDatabaseStats-repo")
+	branchName := "master"
+	numCommits := 5
+	c, _ := minikubetestenv.AcquireCluster(t)
+
+	createTestCommits(t, repoName, branchName, numCommits, c)
+
+	buf := &bytes.Buffer{}
+	filter := &debug.Filter{
+		Filter: &debug.Filter_Database{Database: true},
+	}
+
+	time.Sleep(5 * time.Second) // give some time for the stats collector to run.
+
+	require.NoError(t, c.Dump(filter, 100, buf), "dumping database files should succeed")
+
+	gr, err := gzip.NewReader(buf)
+	require.NoError(t, err)
+
+	require.NoError(t, tarutil.Iterate(gr, func(f tarutil.File) error {
+		fileContents := &bytes.Buffer{}
+		if err := f.Content(fileContents); err != nil {
+			return errors.EnsureStack(err)
+		}
+
+		hdr, err := f.Header()
+		require.NoError(t, err, "getting database tar file header should succeed")
+
+		require.NotMatch(t, "^[a-zA-Z0-9_\\-\\/]+\\.error$", hdr.Name)
+
+		if hdr.Name == "database/row-counts.json" {
+			validateRowCounts(t, *fileContents)
+		}
+
+		return nil
+	}))
+}
+
+func createTestCommits(t *testing.T, repoName, branchName string, numCommits int, c *client.APIClient) {
+	require.NoError(t, c.CreateRepo(repoName), "pods should be available to create a repo against")
+
+	for i := 0; i < numCommits; i++ {
+		commit, err := c.StartCommit(repoName, branchName)
+		require.NoError(t, err, "creating a test commit should succeed")
+
+		err = c.PutFile(commit, "/file", bytes.NewBufferString("file contents"))
+		require.NoError(t, err, "should be able to add a file to a commit")
+
+		err = c.FinishCommit(repoName, branchName, commit.ID)
+		require.NoError(t, err, "finishing a commit should succeed")
+	}
+}
+
+func validateRowCounts(t *testing.T, buf bytes.Buffer) {
```

Comment:
could you move this function definition into TestDatabaseStats since it's only being used in one place. The only reason it's advisable here is because this file has SO many tests it's nice when most of the test definition is in one place.

Reply:
Yup, for this file that makes sense